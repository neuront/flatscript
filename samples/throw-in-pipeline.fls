x: [0, 1, 2, 3]

try
    x |:
        if $ % 4 = 3
            throw 'sync pipeline throw:' + $
catch
    console.log($e)

try
    x |:
        setTimeout(%, 1)
        if $ % 4 = 3
            throw 'async pipeline throw:' + $
catch
    console.log($e)

func aGet(n, %%)
    if n % 3 = 2
        throw 'aGet throw:' + n

func sGet(n)
    if n % 3 = 2
        throw 'sGet throw:' + n

func piping(%%)
    try
        x |: aGet($, %%)
    catch
        console.log($e)
    try
        x |: sGet($)
    catch
        console.log($e)

piping(%%)
console.log('end')
