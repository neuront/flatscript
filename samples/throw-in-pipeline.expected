sync pipeline throw:3
async pipeline throw:3
aGet throw:2
sGet throw:2
end
